### PR TITLE
New vault details section

### PR DESCRIPTION
**Pull request by piotrkonowrocki**
# New vault details section

This story PR consists of three stories, they couldn't be tested separately.
- https://app.shortcut.com/oazo-apps/story/3857/redesign-update-detail-section
- https://app.shortcut.com/oazo-apps/story/3856/redesign-update-detail-cards
- https://app.shortcut.com/oazo-apps/story/4134/redesign-update-detail-section-footer
  
## Changes 👷‍♀️

image

New components created for this view:
- `DetailsSection`,
- `DetailsSectionContentCard`,
- `DetailsSectionFooterItem`,
- `ExpandableButton`,
- `ExpandableArrow`.

All of those components are meant to be used in new layout of vault on every tab. Although, `ExpandableArrow` is little animation component with up/down arrow that could be use in every place that has expandable secions like FAQ, vault history or language selector. This animation (and rest of them) were discussed with both Henry and Hazel.

Also, modal and hover works as intended, Henry told me that hover should trigger only when mouse is over the text, not entire box. That allows this component to have both modal and link below.
  
## How to test 🧪

According to the story, new components shouldn't be implemented yet, the requirements were just for creating them. To test this one would need to pull on this branch and paste code below into `ManageVaultContainer.tsx` file next to (or instead of). `ManageVaultDetails`. Then new components can be tested on borrow vault page.

```
<DetailsSection
  title="Overview"
  badge={true}
  buttons={[
    {
      label: 'Close Vault',
      action: () => {
        alert('Clicked "Close Vault" button!')
      },
    },
    {
      label: 'Edit Position',
      actions: [
        {
          label: 'Edit DAI',
          action: () => {
            alert('Clicked "Edit DAI" option!')
          },
        },
        {
          label: 'Edit Collateral',
          action: () => {
            alert('Clicked "Edit Collateral" option!')
          },
        },
      ],
    },
  ]}
  content={
    <DetailsSectionContentCardWrapper>
      <DetailsSectionContentCard
        title="Liquidation Price"
        value="$1,320.2000"
        footnote="22.34% below current price"
        modal="Content in modal, please fill me somewhere later..."
      />
      <DetailsSectionContentCard
        title="Collateral Ratio"
        value="220.40%"
        footnote="219.10% on next price"
        modal="Content in modal, please fill me somewhere later..."
      />
      <DetailsSectionContentCard
        title="Collateral Locked"
        value="420,000.20"
        unit="DAI"
        change={{ value: '424,602.40 DAI after', variant: 'positive' }}
        footnote="300.30 ETH"
        modal="Content in modal, please fill me somewhere later..."
      />
      <DetailsSectionContentCard
        title="Title"
        link={{
          label: 'Enable now',
          action: () => {
            alert('Clicked "Enable now" link!')
          },
        }}
        modal="Content in modal, please fill me somewhere later..."
      />
    </DetailsSectionContentCardWrapper>
  }
  footer={
    <DetailsSectionFooterItemWrapper>
      <DetailsSectionFooterItem title="Vault DAI Debt" value="103,203.20 DAI" />
      <DetailsSectionFooterItem title="Available to Withdraw" value="44.00 ETH" />
      <DetailsSectionFooterItem title="Available to Generate" value="440.40 ETH" />
      <DetailsSectionFooterItem title="Days Open" value="1130" />
    </DetailsSectionFooterItemWrapper>
  }
/>
```

Please note that `DetailsSection` and `DetailsSectionContentCard` has many props and most of them are optional, so it would be good to try testing all possible scenarios where some of them are filled and some are not. I tried that, but there's always possibility, that I missed some cases.

### CODE REVIEW

File: components/DetailsSectionFooterItem.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  value: string
+}
+
+export function DetailsSectionFooterItemWrapper({ children }: { children: ReactNode }) {
+  return (
+    <Flex
+      as="ul"
+      sx={{
+        mx: -1,
+        mt: -3,
```
Comment (piekczyk) [inline components/DetailsSectionFooterItem.tsx:15]: looks weird, but if it works it's okay

**Review by piekczyk — APPROVED** [review]

File: components/DetailsSectionFooterItem.tsx (hunk 1 above)
Comment (piotrkonowrocki) [reply components/DetailsSectionFooterItem.tsx:15]: ¯\_(ツ)_/¯

File: components/DetailsSection.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
+import React from 'react'
+import { Box, Card, Flex } from 'theme-ui'
+
+import { ExpandableButton, IButtonWithAction, IButtonWithActions } from './ExpandableButton'
+import { VaultTabTag } from './vault/VaultTabTag'
+
+interface IDetailsSectionProps {
```
[the remark this answers is not in the document]
Comment (piotrkonowrocki) [reply components/DetailsSection.tsx:7]: That's just how I've always been doing, interfaces for components were named `I` `NameOfTheComponent` `Props`. How should I change it to match current naming convention?

[the remark this answers is not in the document]
Comment (piekczyk) [reply components/DetailsSection.tsx:7]: Just remove `I` from interface name, so `DetailsSectionProps`

**Review by johnnieskywalker — DISMISSED** [review]
image

image

image


Component looks and behaves as expected

File: components/DetailsSection.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (piotrkonowrocki) [reply components/DetailsSection.tsx]: Done.

**Review by piekczyk — APPROVED** [review]

**Comment by paszkowskiDamian** [comment]
Looks good 👍  

For such scenarios I think that we should keep storybook by the way 😉

**Review by paszkowskiDamian — APPROVED** [review]

**Pull request merged**